sql/descriptor_cache_test.go

```diff
@@ -1,244 +0,0 @@
-// Copyright 2015 The Cockroach Authors.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
-// implied. See the License for the specific language governing
-// permissions and limitations under the License. See the AUTHORS file
-// for names of contributors.
-//
-// Author: Marc Berhault (marc@cockroachlabs.com)
-
-package sql_test
-
-import (
-	"testing"
-	"time"
-
-	"github.com/cockroachdb/cockroach/client"
-	"github.com/cockroachdb/cockroach/config"
-	"github.com/cockroachdb/cockroach/gossip"
-	"github.com/cockroachdb/cockroach/keys"
-	"github.com/cockroachdb/cockroach/server"
-	"github.com/cockroachdb/cockroach/sql"
-	"github.com/cockroachdb/cockroach/util"
-	"github.com/cockroachdb/cockroach/util/leaktest"
-	"github.com/gogo/protobuf/proto"
-)
-
-var configID = sql.ID(1)
-var configDescKey = sql.MakeDescMetadataKey(keys.MaxReservedDescID)
-
-// forceNewConfig forces a system config update by writing a bogus descriptor with an
-// incremented value inside. It then repeatedly fetches the gossip config until the
-// just-written descriptor is found.
-func forceNewConfig(t *testing.T, s *server.TestServer) (*config.SystemConfig, error) {
-	configID++
-	configDesc := sql.DatabaseDescriptor{
-		Name:       "sentinel",
-		ID:         configID,
-		Privileges: &sql.PrivilegeDescriptor{},
-	}
-
-	// This needs to be done in a transaction with the system trigger set.
-	if err := s.DB().Txn(func(txn *client.Txn) error {
-		txn.SetSystemDBTrigger()
-		return txn.Put(configDescKey, &configDesc)
-	}); err != nil {
-		t.Fatal(err)
-	}
-	return waitForConfigChange(t, s)
-}
-
-func waitForConfigChange(t *testing.T, s *server.TestServer) (*config.SystemConfig, error) {
-	var foundDesc sql.DatabaseDescriptor
-	var cfg *config.SystemConfig
-	return cfg, util.IsTrueWithin(func() bool {
-		if cfg = s.Gossip().GetSystemConfig(); cfg != nil {
-			if val := cfg.GetValue(configDescKey); val != nil {
-				if err := val.GetProto(&foundDesc); err != nil {
-					t.Fatal(err)
-				}
-				return foundDesc.ID == configID
-			}
-		}
-
-		return false
-	}, 10*time.Second)
-}
-
-// TestDescriptorCacheSchemaMutation shows how the cache can hide
-// failures when mutating the schema and operating on it.
-// When the operation adds a new value, this works fine since the
-// cache descriptor tries on KV. However, entries that exist in the
-// cache but not in KV will behave badly.
-// This always fails with transactions. Without transaction, it is
-// too dependent on timing to test reliable.
-func TestDescriptorCacheSchemaMutation(t *testing.T) {
```

Comment:
There might be something here that can be revived. I deleted this part way through development of this PR because it kept getting in the way of changes and failing.